tokens/tokens.md

```diff
@@ -0,0 +1,207 @@
+controlled exchange rate between tokens and aeons. Fees cannot be
+payed by native tokens, but only by aeons. Other services could be
+payed for with tokens (e.g., contracts accepting tokens as
+payment). (TODO: Design decision: Should oracles be allowed to set
+fees in tokens?)
+
+The owner of a token can decide to govern the behavior of the tokens
+by connecting a contract when an ANT is created. The contract must
+provide entrypoints according to a specified ACI where one or many of
+the primitive operation (spend, trade, mint, etc) is called to decide
+if the operation is allowed or not.
+
+## Definitions
+
+Note that when *account* is used below, the account can belong to a
+contract, the account may be an oracle, etc.
+
+An Aeternity Native Token (ANT):
+- is a *specification* of tokens
+- is a *first class object* on the chain with its own state tree
+- is *created* by an account which becomes its owner
+- *governs the behavior* when interacting with tokens (trading, sending, minting, etc).
+- can have a *connected contract* to govern the handling of its tokens
+
+A native token:
+- has an ANT describing its behavior.
+- is *countable*
+- is recorded as a *balance in the account* that owns it (in the accounts state tree).
+- is *mintable* (`mint`)
+- is *transferable* (`spend`, `trade`)
+- is *destroyable* (`burn`)
+
+## Tokens state tree
+
+The tokens state tree contains ANT objects. Tokens are stored in the
+accounts state tree in the owner's account.
+
+### ANT object
+
+The token id is determined by the creator's account and nonce.
+```
+  id := Blake2b(<CreatorPubkey><CreateTxNonce>)
+```
+where `Blake2b` is the 256 bit hash. The [API
+serialization](../node/api/api_encoding.md) of an ANT id is tagged by
+`nt_`
+
+The ANT contains the fields:
+- `creator` - The account id of the creator
+- `meta_data` - A byte array field, uninterpreted, but under consensus
+- `contract` - A contract id if there is a governing contract, or the empty binary otherwise
+- `total_supply` - A counter of the currently available amount of the token
+- `parent` - The id of the parent ANT (TODO: Hierarchical tokens?)
+- `final` - If the ANT is final, new tokens cannot be minted. Can flip to true, but never back to false again.
+
+The `meta data` is an uninterpreted string but token minters are
+encouraged to use the following json object: (TODO: Extend).  The
+intention of the meta data is to have a consensus controlled way of
+providing information usable by tools (e.g., displaying an intended
+denominator, a display name, etc).
+```
+{
+  "type" : "object",
+  "properties": { "name": {"type" : "string"},
+                }
+}
+```
+
+A contract governing the usage of tokens must have a non-empty subset
+of the following ACI:
+
+```
+ spend(recipient : address, payload : Type) : boolean
+ trade([(from : address, to: address, Option(token : address))], payload : Type) : boolean
+ mint(amount: integer) : boolean
+ burn(amount : integer) : boolean
```

Comment:
`Call.caller` = the account burning the tokens, presumably. And same for `spend`.